### PR TITLE
Add comment document to fontFamily in the IconData

### LINKED ISSUE #154149: Add an example comment for fontFamily in IconData as hint
### Use case

There was an [issue](https://github.com/flutter/flutter/issues/154144) about the `codePoint` not working. It was caused by missing the value of fontFamily in Icon. Although it is rare to use it directly, It is still very common and intuitive mistakes. So I suggest to add some comment on the document to hint developer.

### Proposal

I suggest to add these comment on IconData.
Screenshot 2024-08-27 at 1 31 44 AM

```Dart
  /// The [fontFamily] argument is normally required when using custom icons.
  /// 
  /// e.g. When you are using the [codePoint] in 
  /// [Material Icons](https://material.io/resources/icons), You should specify
  /// that `fontFamily: 'MaterialIcons'`. See the source code in [Icons].
  /// 
  /// e.g. When you are using the [codePoint] in your [CustomIcons]
  /// ```yaml
  /// fonts:
  ///   - family: CustomIcons
  ///     fonts:
  ///       - asset: assets/fonts/CustomIcons.ttf
  /// ```
  /// You should specify that `fontFamily: 'CustomIcons'`
```

And I have created an [PR](https://github.com/flutter/flutter/pull/154150/files) with these change.

### PR DESCRIPTION
**Pull request by SuicaLondon**
I’ve added a comment document for the `fontFamily` property of `IconData`. The purpose of this is to provide developers with a hint when importing a custom icon. It only includes the change made in the comment.

The original issue was: 
[154144](https://github.com/flutter/flutter/issues/154144)
It was marked as closed so I created a new issues for proposal:
Fixes #154149

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Review by guidezpl — CHANGES_REQUESTED** [review]

**Review by guidezpl — APPROVED** [review]

File: packages/flutter/lib/src/widgets/icon_data.dart [hunk 1]
```diff
@@ -25,7 +25,22 @@ class IconData {
   ///
   /// Rarely used directly. Instead, consider using one of the predefined icons
   /// like the [Icons] collection.
-  ///
+  /// 
+  /// The [fontFamily] argument is normally required when using custom icons.
+  /// 
+  /// e.g. When you are using the [codePoint] in 
+  /// [Material Icons](https://material.io/resources/icons), You should specify
+  /// that `fontFamily: 'MaterialIcons'`. See the source code in [Icons].
```
[the remark this answers is not in the document]
Comment (guidezpl) [reply packages/flutter/lib/src/widgets/icon_data.dart]: L26 mentions [Icons]

**Comment by guidezpl** [comment]
You'll also want to remove trailing spaces

File: packages/flutter/lib/src/widgets/icon_data.dart [hunk 2]
```diff
@@ -25,7 +25,18 @@ class IconData {
   ///
   /// Rarely used directly. Instead, consider using one of the predefined icons
   /// like the [Icons] collection.
-  ///
+  /// 
+  /// The [fontFamily] argument is normally required when using custom icons.
+  /// 
+  /// e.g. When you are using the [codePoint] in your [CustomIcons]
+  /// ```yaml
```
[the remark this answers is not in the document]
Comment (SuicaLondon) [reply packages/flutter/lib/src/widgets/icon_data.dart]: Thank you so much for the suggestion, I have committed these change.

File: packages/flutter/lib/src/widgets/icon_data.dart [hunk 3]
```diff
@@ -25,7 +25,18 @@ class IconData {
...
+  /// fonts:
+  ///   - family: CustomIcons
+  ///     fonts:
+  ///       - asset: assets/fonts/CustomIcons.ttf
+  /// ```
+  /// You should specify that `fontFamily: 'CustomIcons'`
```
[the remark this answers is not in the document]
Comment (SuicaLondon) [reply packages/flutter/lib/src/widgets/icon_data.dart]: Thank you so much for the suggestion, I also have committed this change.

**Comment by SuicaLondon** [comment]
> You'll also want to remove trailing spaces

Thank you very much, I have removed all trailing spaces.

**Comment by guidezpl** [comment]
Thanks!

**Review by piedcipher — APPROVED** [review]
LGTM

**Pull request merged**